src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php

```diff
@@ -123,12 +123,13 @@ public static function register($mode = 0)
             // triggered deprecation notice; and the fourth event (index 3)
             // represents the action that called the deprecated code. In the
             // scenario that we want to suppress, the 4th event will be an
-            // object instance of \PHPUnit\Framework\TestCase.
+            // extension of Symfony\Bundle\FrameworkBundle\Test\KernelTestCase.
             if (isset($trace[3]['object'])) {
                 $isPrivateServiceNotice = false !== strpos($msg, ' service is private, ');
                 $isNoticeForContainerGetHasUsage = 'Symfony\Component\DependencyInjection\Container' === $trace[2]['class'] && in_array($trace[2]['function'], array('get', 'has'));
-                $noticeWasTriggeredByPhpUnitTest = $trace[3]['object'] instanceof \PHPUnit\Framework\TestCase;
-                if ($isPrivateServiceNotice && $isNoticeForContainerGetHasUsage && $noticeWasTriggeredByPhpUnitTest) {
+                $kernelTestCaseClass = 'Symfony\\Bundle\\FrameworkBundle\\Test\\KernelTestCase';
```

Comment:
this var should be removed, instanceof works better with a symbol than with a string

Reply:
(and a `use` statement be used instead)

Reply:
this should definitely be WebTestCase at least, as KernelTestCase only allows to use `$kernel->getContainer()->get('private_something')` in 3.x, which is always a valid deprecation